apis/kueue/webhooks/workload_webhook.go

```diff
@@ -57,10 +58,28 @@ func (w *WorkloadWebhook) Default(ctx context.Context, obj runtime.Object) error
 		if len(podSet.Name) == 0 {
 			podSet.Name = kueue.DefaultPodSetName
 		}
+		setContainersDefaults(podSet.Spec.InitContainers)
+		setContainersDefaults(podSet.Spec.Containers)
 	}
 	return nil
 }
 
+func setContainersDefaults(containers []corev1.Container) {
+	for i := range containers {
+		c := &containers[i]
+		if c.Resources.Limits != nil {
```

Comment:
Not related to this PR, we currently leverage the requests in resource quota, should we also consider the limits.

Reply:
Are you talking about kubernetes' ResourceQuotas? If so, I would imagine they run after defaults are applied, but I don't have expertise to know for sure.

Reply:
I mean the resources.Limits in podSpec.

Reply:
It was the intention that we only look at requests, similar to the NodeResources Filter plugin in kube-scheduler.